src/user/indexable_functions.cpp

```diff
@@ -3,501 +3,111 @@
-                                    complementaryColor, 500, isFinished, strip)
-                              : animations::color_wipe_down(
-                                    complementaryColor, 500, isFinished, strip);
-      if (isFinished) {
-        switchMode = !switchMode;
-        complementaryColor.update();  // update color
-      }
-      break;
-
-    case 1:
-      // random solid color
-      static GenerateRandomColor randomColor = GenerateRandomColor();
-      if (categoryChange) randomColor.reset();
-
-      isFinished =
-          animations::double_side_fill(randomColor, 500, isFinished, strip);
-      if (isFinished) randomColor.update();  // update color
-      break;
-
-    case 2:
-      static GenerateComplementaryColor complementaryPingPongColor =
-          GenerateComplementaryColor(0.4);
-      if (categoryChange) complementaryPingPongColor.reset();
-
-      // ping pong a color for infinity
-      isFinished = animations::dot_ping_pong(complementaryPingPongColor, 1000.0,
-                                             128, isFinished, strip);
-      if (isFinished) complementaryPingPongColor.update();  // update color
-      break;
-
-    default:  // error
-      colorState = 0;
-      strip.clear();
-      break;
-  }
-
-  // reset category change
-  categoryChange = false;
-}
+#else
+#warning "This file requires --std=gnu++17 or higher to build!*"
+//
+// *if you got this warning, check Makefile to see how to build project
 
-void sound_mode_update() {
-  constexpr uint8_t maxSoundState = 0;
-  switch (clamp_state_values(colorState, maxSoundState)) {
-    default:  // error
-      colorState = 0;
-      strip.clear();
-      break;
-  }
-
-  // reset category change
-  categoryChange = false;
-}
+//
+// no c++17 support -- placeholder implementation
+//
 
-void sirenes_mode_update() {
-  constexpr uint8_t maxGyroState = 0;
-  switch (clamp_state_values(colorState, maxGyroState)) {
-    case 0:
-      animations::police(500, false, strip);
-      break;
-
-    default:  // error
-      colorState = 0;
-      strip.clear();
-      break;
-  }
-
-  // reset category change
-  categoryChange = false;
-}
+namespace user {
 
-void color_mode_update() {
-  constexpr uint8_t maxColorMode = 4;
-  switch (clamp_state_values(colorMode, maxColorMode)) {
-    case 0:  // Fixed colors, that the user can change
-      gradient_mode_update();
-      break;
-
-    case 1:  // slow changing animations, nice to look at
-      calm_mode_update();
-      break;
-
-    case 2:
-      // fast pacing animations
-      party_mode_update();
-      break;
-
-      /*
-        case 3:  // sound reacting mode
-          sound_mode_update();
-          break;
-      */
-
-      /*
-      case 4:  // diverse alerts (police, firefighters etc)
-        sirenes_mode_update();
-        break;
-      */
-
-    default:
-      colorMode = 0;
-      break;
-  }
-
-  // reset mode change
-  modeChange = false;
-}
+LedStrip strip(AD0);
```

Comment:
It should be the same object as the one in _private, or we will instanciate multiple buffers

Reply:
This code is in the `#else` condition of the `#ifdef LMBD_CPP17` guarding the inclusion of the C++17 code above. The content of this bottom section of `src/user/indexable_functions.cpp` is thus a placeholder implementation, not included in the final code, and basically a stripped down version of the previously existing `src/user/indexable_functions.cpp` code.

This `LedStrip strip(AD0)` is thus never instantiated, as this code is not included during the final binary build. However, as a caveat we still need the project to build properly without `LMBD_CPP17` (because of the hack in the build system) hence the presence of that placeholder.

I've tested with the following change and it seems it still build fine:
```diff
diff --git a/src/user/indexable_functions.cpp b/src/user/indexable_functions.cpp
index 436d113..0f847fb 100644
--- a/src/user/indexable_functions.cpp
+++ b/src/user/indexable_functions.cpp
@@ -67,26 +67,8 @@ static constexpr uint32_t LED_POWER_PIN = AD1;

 namespace user {

-LedStrip strip(AD0);
-static constexpr uint32_t LED_POWER_PIN = AD1;
-
-void power_on_sequence() {
-  pinMode(LED_POWER_PIN, OUTPUT);
-  digitalWrite(LED_POWER_PIN, HIGH);
-
-  strip.begin();
-  strip.clear();
-  strip.show();
-  strip.setBrightness(BRIGHTNESS);
-}
-
-void power_off_sequence() {
-  strip.clear();
-  strip.show();
-
-  digitalWrite(LED_POWER_PIN, LOW);
-  pinMode(LED_POWER_PIN, OUTPUT_H0H1);
-}
+void power_on_sequence() { }
+void power_off_sequence() { }

 void brightness_update(const uint8_t) { }
 void write_parameters() { }
 ```
 
 I hope this clarify the situation, I'll apply this and move forward, marking this as solved.